docs/compute-resources.md

```diff
@@ -52,6 +52,251 @@ requirements for `Step`containers, they must be treated as if they are running i
+
+### Interaction with Step-level Resources Requirements
+
+When Tekton now supports configuring step-level resources requirements, from `Task.spec.step.resources`, `Task.spec.stepTemplate.resources`, `TaskRun.spec.stepOverrides.resources`, and `PipelineRun.spec.taskRunSpecs.stepOverrides.resources`, either the task-level or step-level will be allowed to configure resources requirements, therefore avoiding possible conflicts.
+
+The following configurations will be rejected during the webhook validation:
+
+Case #1: `Task.spec.step.resources` and `Task.spec.resources`
+
+```yaml
+kind: Task
+spec:
+  steps:
+    - name: foo
+      requests:
+        cpu: 1
+  resources:
+    requests:
+      cpu: 2
+```
+
+Case #2: `Task.spec.stepTemplate.resources` and `Task.spec.resources`
+
+```yaml
+kind: Task
+spec:
+  steps:
+    - name: foo
+  stepTemplate:
+    requests:
+      cpu: 1
+  resources:
+    requests:
+      cpu: 2
+```
+
+Case #3: `TaskRun.spec.stepOverrides.resources` and `TaskRun.spec.taskSpec.resources`
+
+```yaml
+kind: TaskRun
+spec:
+  stepOverrides:
+    - name: foo
+      resources:
+        requests:
+          cpu: 1
+  taskSpec:
+    resources:
+      requests:
+        cpu: 2  
+```
+
+Case #4: `PipelineRun.spec.taskRunSpecs.stepOverrides.resources` and `PipelineRun.spec.taskRunSpecs.resources`
+
+```yaml
+kind: PipelineRun
+spec:
+  taskRunSpecs:
+    - pipelineTaskName: foo
+      stepOverrides:
+        - name: foo 
+          resources:
+            requests:
+              cpu: 1 
+      resources:
+        requests:
+          cpu: 2 
+```
+
+### Interaction with Runtime Overwriting
+
+As runtime specifications are based on an execution context, Tekton allows users to overwrite certain statically configured fields in the runtime. For task-level resources requirements, there are `TaskRun.spec.taskSpec.resources` and `PipelineRun.spec.taskRunSpecs.resources`.
+
+e.g.
+
+```yaml
+kind: Task
+metadata:
+  name: foo
+spec:
+  resources:
+    requests:
+      cpu: 1
+---
+kind: TaskRun
+spec:
+  taskRef:
+    name: foo
+  taskSpec:
+    resources:
+      requests:
+        cpu: 2
+```
+
+The final task cpu resources request will be 2 by the overwriting.
+
+### Interaction with LimitRanges
+
+Min or max resources requirements can be configured by `LimitRange`. For a min request on certain containers, the task-level request can subtract this amount and assign the rest to one `Step`(container) as stated before. Furthermore, if a container has a value more than the max limit after being set with task-level requests, the request amount will be spread out to each container (the number will be rounded to ignore decimals).
```

Comment:
Some of this also gets into implementation details. Since there are multiple possible ways to configure container resource requirements to meet task level resource requirements, I wouldn't describe which way we're choosing in the docs.

Reply:
(as a previous comment) let me refine around more under a user doc view. 

thanks this point!